### DISCUSSION TITLE
flux bootstrap github failing

### DISCUSSION THREAD

**Discussion by smartlizzard** [fluxcd/flux2#2929] [General]
I am using EKS cluster and trying to implement GitOps using fluxcd. When I am trying to bootstrap with my GitHub repository getting error.
Commands:
flux bootstrap github --owner=xxxxxxxxxx --repository=fluxcd --path=clusters/eskool-prod-cluster --personal

output:

► connecting to github.com
► cloning branch "main" from Git repository "https://github.com/xxxxxxxxx/fluxcd.git"
✔ cloned repository
► generating component manifests
✔ generated component manifests
✔ component manifests are up to date
► installing components in "flux-system" namespace
✔ installed components
✔ reconciled components
► determining if source secret "flux-system/flux-system" exists
✔ source secret up to date
► generating sync manifests
✔ generated sync manifests
✔ sync manifests are up to date
► applying sync manifests
✔ reconciled sync configuration
◎ waiting for Kustomization "flux-system/flux-system" to be reconciled
✗ Get "https://xxxxxxxxxxxxxxxxx.gr7.us-east-1.eks.amazonaws.com/apis/kustomize.toolkit.fluxcd.io/v1beta2/namespaces/flux-system/kustomizations/flux-system": context deadline exceeded
► confirming components are healthy
✔ helm-controller: deployment ready
✔ kustomize-controller: deployment ready
✔ notification-controller: deployment ready
✔ source-controller: deployment ready
✔ all components are healthy
✗ bootstrap failed with 1 health check failure(s)

K8s version: 1.22

**Comment by kingdonb** [comment]
This looks like a network issue, if the Flux pods cannot reach the EKS control plane then it will not function. I would guess that health check failure is somehow related to a CNI failure, it could possibly be an error of leader election or some other communication between the Flux pods that is failing.

Do you have any way of verifying the CNI? You should be able to tell from a pod within the `flux-system` namespace that you can reach the internet (github.com) and that pods can communicate with each other as well as the control plane. This might be due to security groups.

Can you provide any more info or did you manage to resolve this?

It could also be an issue with the Network Policy resources. Another suggestion to try is to remove the network policy resources with `--network-policy=false` on the bootstrap command.
